toucan_connectors/utils/pem.py

```diff
@@ -4,28 +4,13 @@ def __init__(self):
 
 
 def _sanitize_spaces_pem(pem_data: str) -> str:
-    it = enumerate(pem_data)
-    # Can't use tuples in walrus assignments :-(
-    # Skipping the header between --- and letting it untouched
-    while next(it)[1] == '-':
-        continue
-    while next(it)[1] != '-':
-        continue
-    while (tup := next(it))[1] == '-':
-        continue
+    pem_data = pem_data.replace('-----BEGIN CERTIFICATE-----', '-----BEGIN_CERTIFICATE-----')
+    pem_data = pem_data.replace('-----END CERTIFICATE-----', '-----END_CERTIFICATE-----')
+    pem_data = pem_data.replace(' ', '\n')
+    pem_data = pem_data.replace('-----BEGIN_CERTIFICATE-----', '-----BEGIN CERTIFICATE-----')
+    pem_data = pem_data.replace('-----END_CERTIFICATE-----', '-----END CERTIFICATE-----')
 
```

Comment:
This does not work with all PEM data: it might be `BEGIN RSA PRIVATE KEY` / `BEGIN PRIVATE KEY` / `BEGIN CERTIFICATE` / `BEGIN RSA PUBLIC KEY` and way more... it's not a good format :/

Reply:
I fear you need to keep this logic, but just not for the first block... I can have a look if you want